src/prefect/runner/storage.py

```diff
@@ -242,21 +282,39 @@ async def _clone_repo(self):
         ]
         if self._branch:
             cmd += ["--branch", self._branch]
-        if self._include_submodules:
-            cmd += ["--recurse-submodules"]
 
-        # Limit git history and set path to clone to
-        cmd += ["--depth", "1", str(self.destination)]
+        # For sparse-checkout it is recommended to use --filter=blob:none to reduce disk-space
+        if self._sparse_checkout_mode:
```

Comment:
I put this in the else block as I was unsure what submodules did, but are they something to do with adding code from another repo into the cloned repo?

I think this would be easy with cone mode but might be trickier without.
Would it be useful to also sparse-checkout the submodule?

Reply:
Yeah, submodules allow you to reference another git repository from a parent repository.  I'm not sure how sparse checkout would interact with submodules, but I think it'd be useful to apply sparse checkout to submodules, also.